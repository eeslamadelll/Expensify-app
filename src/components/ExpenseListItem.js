import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id, description, amount, createdAt}) => {
    return (
        <div>
            <Link to={`/edit/${id}`}>
                <p>Description : { description }</p>
            </Link>
            <span>Created At : { moment(createdAt).format('MMMM Do, YYYY') }</span>
            <p>Amount : { numeral(amount / 100).format('$0,0.00') }</p>
        </div>
    )
};

export default ExpenseListItem;