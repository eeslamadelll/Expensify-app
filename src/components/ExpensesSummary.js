import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';


export const ExpensesSummary = ({expenseCount, expenseTotal}) => {
    const expenseWord = expenseCount > 1 ? 'expenses' : 'expense';
    return(
        <div className="page-header">
            <div className="content-container">
                <p>you have <span>{expenseCount}</span> {expenseWord} with total of <span>{numeral(expenseTotal / 100).format('$0,0.00')}</span></p>
                <Link className="add" to="/add">Add Expense</Link>
            </div>
        </div>
    );
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: getExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);