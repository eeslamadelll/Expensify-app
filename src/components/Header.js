import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => {
    return (
        <header>
            <h1>Expensify</h1>
            <nav>
                <ul>
                    <li><NavLink to="/dashboard" activeClassName="is-active">Homepage</NavLink></li>
                    <li><NavLink to="/add" activeClassName="is-active">Add Expense</NavLink></li>
                </ul>
                <button onClick={props.startLogout}>Logout</button>
            </nav>
        </header>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
}

export default connect(undefined, mapDispatchToProps)(Header);