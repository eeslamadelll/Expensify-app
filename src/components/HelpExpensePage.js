import React from 'react';

const HelpExpensePage = () => {
    return (
        <div>
            Help
        </div>
    );
}

export default HelpExpensePage;