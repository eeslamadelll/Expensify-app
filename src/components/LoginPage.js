import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = (props) => {
    return (
        <div className="login">
            <div className="overlay"></div>
            <div className="container">
                <h1>Expensify App</h1>
                <p>It's time to put your expenses under control</p>
                <button onClick={props.startLogin}>Login with Google</button>
            </div>
        </div>
    );
}

const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: () => dispatch(startLogin())
    }
}

export default connect(undefined, mapDispatchToProps)(LoginPage);