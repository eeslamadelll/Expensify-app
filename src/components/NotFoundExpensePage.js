import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundExpensePage = () => {
    return (
        <div>
            404 Not Found Page! <Link to="/">Back Home</Link>
        </div>
    );
}

export default NotFoundExpensePage;