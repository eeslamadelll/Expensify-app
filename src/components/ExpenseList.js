import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
    	{
    		props.expenses.length === 0 ? (
    			<p>There is no expenses to show</p>
    		) : (
    			props.expenses.map((expense) => {
			        return <ExpenseListItem key={expense.id} {...expense}/>;
			    })
    		)
    	}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList);


///////////////////// this code is right but we will use the best practice
// const ConnectExpenseList = connect((state) => {
//     return {
//         expenses: state.expenses,
//         filters: state.filters
//     };
// })(ExpenseList);

// export default ConnectExpenseList;