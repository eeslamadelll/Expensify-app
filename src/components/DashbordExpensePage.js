import React from 'react';
import ExpenseList from './ExpenseList'; // data comes from HOC
import ExpenseListFilters from './ExpenseListFilters';

const DashbordExpensePage = () => {
    return (
        <div>
            <ExpenseListFilters />
            <ExpenseList />
        </div>
    );
}

export default DashbordExpensePage;