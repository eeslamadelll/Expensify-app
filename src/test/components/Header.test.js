import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

test('Should render header component correctly', () => {
    const wrapper = shallow(<Header startLogout={() => { }}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render header and logout button behaviour', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});