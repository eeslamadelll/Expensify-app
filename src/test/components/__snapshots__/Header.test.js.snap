// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header component correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav>
    <ul>
      <li>
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          exact={true}
          to="/"
        >
          Homepage
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          to="/add"
        >
          Add Expense
        </NavLink>
      </li>
    </ul>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </nav>
</header>
`;
